### PR TITLE
A few updates following discussions on ruby-docs-samples

### PR DESCRIPTION
**Pull request by TheRoyalTnetennba**
https://github.com/GoogleCloudPlatform/ruby-docs-samples/pull/408 for referenced discussion

* Enforce expaned for Style/EmptyMethod
* Enforce braces_for_chaining for Style/BlockDelimiters
* Allow parenthesis in chaining for Style/MethodCallWithArgsParentheses

* Would also be curious to hear everyone's thoughts on:
  - Enforce require_no_parentheses_except_multiline for Style/MethodDefParentheses
  - Enforce brackets for Style/SymbolArray and Style/WordArray (so we get `[:symbol, :other_symbol]` instead of `%i[foo bar baz]`

### CODE REVIEW

File: google-style.yml [hunk 1]
```diff
@@ -35,13 +35,18 @@ Metrics/PerceivedComplexity:
   Max: 10
 Naming/VariableNumber:
   Enabled: false
+Style/BlockDelimiters:
+  EnforcedStyle: braces_for_chaining
```
Comment (quartzmo) [inline google-style.yml]: Just curious, what are the reasons for the `braces_for_chaining` style?

Comment (TheRoyalTnetennba) [reply google-style.yml]: Changes `datastore.query("Visit").order("timestamp", :desc).limit 10` to `datastore.query("Visit").order("timestamp", :desc).limit(10)`
I'm not super dogmatic about it one way or the other, but I prefer the second. Do you prefer the first?

**Comment by quartzmo** [comment]
> Enforce require_no_parentheses_except_multiline for Style/MethodDefParentheses

We've omitted parentheses for multiline parameter lists in so places that I'd rather not switch this rule.

File: google-style.yml (hunk 1 above)
Comment (quartzmo) [reply google-style.yml]: Are you sure you have the right rule? See https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/BlockDelimiters:

```rb
EnforcedStyle: braces_for_chaining
# bad
words.each do |word|
  word.flip.flop
end.join("-")

# good
words.each { |word|
  word.flip.flop
}.join("-")
```

**Comment by quartzmo** [comment]
> Enforce brackets for Style/SymbolArray and Style/WordArray (so we get [:symbol, :other_symbol] instead of %i[foo bar baz]

I can go either way on this one.

File: google-style.yml (hunk 1 above)
Comment (blowmage) [reply google-style.yml]: I prefer `line_count_based` for `BlockDelimiters`. Another approach I've seen is to use curly braces when you care about the return value, and do/end when you don't. I never liked that style myself, but that seems to be what `braces_for_chaining` is specifying.

**Comment by blowmage** [comment]
> Enforce brackets for Style/SymbolArray and Style/WordArray (so we get [:symbol, :other_symbol] instead of %i[foo bar baz]

I agree with Chris. I don't read `%i[foo bar baz]` as naturally as I read `[:foo, :bar, :baz]`, but I understand why someone would push the former. Either is fine to me, but I would give the latter a slight preference for code examples and documentation.

File: google-style.yml [hunk 2]
```diff
@@ -35,13 +35,18 @@ Metrics/PerceivedComplexity:
...
+Style/EmptyMethod:
+  EnforcedStyle: expanded
 Style/FrozenStringLiteralComment:
   Enabled: false
 Style/MethodCallWithArgsParentheses:
   Enabled: true
   EnforcedStyle: omit_parentheses
   AllowParenthesesInMultilineCall: true
   AllowParenthesesInCamelCaseMethod: true
+  AllowParenthesesInChaining: true
```
Comment (blowmage) [inline google-style.yml]: This is the setting that looks to allow `datastore.query("Visit").order("timestamp", :desc).limit(10)`.

File: google-style.yml [hunk 3]
```diff
@@ -35,13 +35,18 @@ Metrics/PerceivedComplexity:
   Max: 10
 Naming/VariableNumber:
   Enabled: false
+Style/BlockDelimiters:
+  EnforcedStyle: braces_for_chaining
+Style/EmptyMethod:
+  EnforcedStyle: expanded
```
Comment (blowmage) [inline google-style.yml]: :+1:

File: google-style.yml (hunk 1 above)
Comment (TheRoyalTnetennba) [reply google-style.yml]: @quartzmo yeah, got them mixed up. I actually don't really have feelings on this one, other than keeping line_count_based enabled

Comment (quartzmo) [reply google-style.yml]: @TheRoyalTnetennba Can you remove `EnforcedStyle: braces_for_chaining` then?

File: google-style.yml [hunk 4]
```diff
@@ -53,5 +53,9 @@ Style/RescueModifier:
   Enabled: false
 Style/StringLiterals:
   EnforcedStyle: double_quotes
+Style/SymbolArray:
+  EnforcedStyle: brackets
```
Comment (blowmage) [inline google-style.yml]: Is this something we want in the global standard rather than having ruby-docs-samples configure an exception for their usage?

Comment (TheRoyalTnetennba) [reply google-style.yml]: I prefer this style. If we all either prefer `[:symbol, :symbol]`, `["string", "string"]` or don't care, we might as well have a uniform approach across repos.

File: google-style.yml (hunk 1 above)
Comment (TheRoyalTnetennba) [reply google-style.yml]: We can. The discussion for ruby-docs-samples seemed to favor `braces_for_chaining`. I'm close to ambivalent. Do you have a preference?

Comment (blowmage) [reply google-style.yml]: I would prefer `line_count_based` and to have ruby-docs-samples override this rule in their own configuration.

**Comment by TheRoyalTnetennba** [comment]
@quartzmo @dazuma are you ok with the changes?

**Pull request merged**